mittab/apps/tab/team_views.py

```diff
@@ -380,3 +380,120 @@ def team_stats(request, team_id):
     return JsonResponse(data)
 
 
+def public_status_by_team(request, team_name, team_ref=None):
+    if team_ref:
+        matches = [team_ref]
+    else:
+        clean = team_name.replace('_', ' ')
+        matches = [*Team.objects.filter(name__iexact=clean)]
+    
+    if matches:
+        round_number = TabSettings.get('cur_round') - 1        
+        pairing_exists = TabSettings.get('pairing_released', 0) == 1
+
+        if pairing_exists:
+            return _public_status_paring_exists(request, team_name, matches, 
+                                                round_number)
+        else:
+            out = {
+                    'no_navigation': True,
+                    'bye': False,
+                    'pairing_exists': False,
+                    'team_name': None,
+                    'gov_or_opp': None,
+                    'other_team': None,
+                    'room': None,
+                    'judges': None,
+                    'round_number': round_number,
+                  }
+
+            return render_to_response('public_status.html', out,
+                                  context_instance=RequestContext(request))
+    else:
+        return render_to_response('error.html',
+                                 {'error_type': 'Tab Card','error_name': 'View',
+                                  'error_info': "That's not a registered team!",
+                                  'no_navigation': True},
+                                  context_instance=RequestContext(request))
+                                  
+
+def _public_status_paring_exists(request, team_name, matches, round_number):
+    # Note here that the queries returned by Model.objects.filter are evaluated
+    # lazily, so there is no performance penalty to defining them all up here
+    gov_pairings = Round.objects.filter(round_number=round_number, 
+                                        gov_team=matches[0])
+    opp_pairings = Round.objects.filter(round_number=round_number, 
+                                        opp_team=matches[0])
+    byes = Bye.objects.filter(round_number=round_number,
+                              bye_team=matches[0])   
+        
+    if gov_pairings:
```

Comment:
perhaps a nitpick, but I'd prefer to move this logic to the template rather than in the view, since a lot of it feels a lot more like "how to I display this thing" sorts of issues. So the logic in the team_views.py would roughly be:

```
if pairing is a round:
	render('public_status', is_round=True, obj=round)
elif pairing is a bye:
	render('public_status', is_round=False, obj=bye)
else:
	render error
```

Then in the view we have an if statements to do the rest of the stuff and keep the `team_views.py` file relatively slim

Reply:
I've moved a lot of the logic to template, but not quite all of it. In particular, it is much easier to process the judge list into a string in `team_views.py`, so I've kept that in there.